nrf-hal-common/src/saadc.rs

```diff
@@ -65,18 +84,29 @@ impl Saadc {
     }
 }
 
+// TODO problems:
+// - other wording than the pac docs that are pulled in in ../ENUMS
+// - default value doc may get out of sync
 pub struct SaadcConfig {
+    #[doc = "Output resolution in bits. Default value: `Resolution::_14BIT`"]
```

Comment:
Same here, you can use `///` on each field.

In general:

* `//!` is used to apply to the top of the current scope. It typically goes at the top of the file. It is usually used for file/library/module scope comments.
* `///` is used to decorate the NEXT item, so it goes above the struct/field/variant/function/etc.